### PR TITLE
fpgad: harden the workspace

### PR DESCRIPTION
**Pull request by tswhison**
Deprecate --directory and --umask parameters.
Restrict --logfile and --pidfile parameters to allowing file names only.
Set directory based on the current user:
* If root, then use /var/lib/opae
* Otherwise, use ${HOME}/.opae
Set the umask based on the current user:
* If root, deny world read/write and group write.
* Otherwise, deny world write and group write.

### CODE REVIEW

File: tools/base/fpgad/fpgad.c [hunk 1]
```diff
@@ -101,6 +109,103 @@ void sig_handler(int sig, siginfo_t *info, void *unused)
...
+		} else {
+			char cwd[PATH_MAX];
+			// ${HOME} not found - use current dir.
+			if (getcwd(cwd, sizeof(cwd))) {
+				snprintf_s_s(c->directory, sizeof(c->directory),
+						"%s/.opae", cwd);
+			} else {
+				// Current directory not found - use /
+				strncpy_s(c->directory, sizeof(c->directory),
+						"/.opae", 6);
```
Comment (zhangzhang10) [inline tools/base/fpgad/fpgad.c]: Does a non-root user have write permission to "/" ?

File: tools/base/fpgad/fpgad.c [hunk 2]
```diff
@@ -101,6 +109,103 @@ void sig_handler(int sig, siginfo_t *info, void *unused)
...
+{
+	char *sub;
+	bool def;
+	mode_t mode;
+	struct stat stat_buf;
+
+	if (!geteuid()) {
+		// If we're being run as root, then use DEFAULT_DIR_ROOT
+		// as the working directory.
+		strncpy_s(c->directory, sizeof(c->directory),
```
Comment (r-rojo) [inline tools/base/fpgad/fpgad.c]: I wonder if this will be flagged by static analysis.

File: tools/base/fpgad/fpgad.c (hunk 1 above)
Comment (tswhison) [reply tools/base/fpgad/fpgad.c]: Probably not, however if the user doesn't have a ${HOME}, chances are it's a privileged / system user.  Worst case, the fopen call will fail.  Would it be better to default to /tmp I wonder?

File: tools/base/fpgad/fpgad.c (hunk 2 above)
Comment (tswhison) [reply tools/base/fpgad/fpgad.c]: How so?

File: tools/base/fpgad/fpgad.c (hunk 1 above)
Comment (zhangzhang10) [reply tools/base/fpgad/fpgad.c]: I guess no need to sweat on this corner case. It should be fine to default to "/".

**Pull request merged**